### PR TITLE
Speed up vector_cast test by using std::async

### PR DESCRIPTION
**Pull request by steven-johnson**
~4x wall-clock improvement on my laptop.

### CODE REVIEW

File: test/correctness/vector_cast.cpp [hunk 1]
```diff
@@ -139,19 +138,24 @@ int main(int argc, char **argv) {
...
+        test_all<uint16_t>(vec_width, target, futures);
+        test_all<int16_t>(vec_width, target, futures);
+        test_all<uint32_t>(vec_width, target, futures);
+        test_all<int32_t>(vec_width, target, futures);
+    }
+
+    bool ok = true;
+    for (auto &f : futures) {
+        f.wait();
+        ok &= f.get();
```
Comment (estollnitz) [inline test/correctness/vector_cast.cpp]: No need to call `wait`, since `get` will wait for the result. Same goes for the `boundary_conditions` test.

**Comment by abadams** [comment]
Please test all these on the windows build bot too. If they work on travis and your laptop, I think that should cover the other targets (e.g. arm)

File: test/correctness/vector_cast.cpp (hunk 1 above)
Comment (steven-johnson) [inline test/correctness/vector_cast.cpp]: Acknowledged.

**Comment by steven-johnson** [comment]
Bringing my local Windows box up to date right now to do in-person testing.

**Comment by steven-johnson** [comment]
Appears that LLVM is broken for Windows in general right now (both on the BuildBot and my local machine.)

**Comment by abadams** [comment]
I've never felt the need to parallelize most of these tests because the build is parallel anyway (I do make -j32). I'm OK with these changes, but I'd like to know why we're doing it. Are these tests are the long pole in a parallel build, are parallel builds worth avoiding, are these tests frequently run in isolation, or is there some other reason?

**Comment by steven-johnson** [comment]
The main reason is that I want to run them inside Google, and a handful of these tests are running past the default test-size time limits on Forge. I could probably special-case them to be size=ginormous, but after I looked inside the tests to see what was going on, my inner obsessive-compulsive took over.

**Comment by steven-johnson** [comment]
Verified on my personal Windows box that these all work (simd_op_check needed minor tweaks, the rest were fine as-is).

**Pull request merged**